components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/BuildTimeMetricsFields.java

```diff
@@ -0,0 +1,48 @@
+package com.gradle.enterprise.cli;
+
+import com.gradle.enterprise.model.BuildTimeMetrics;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public enum BuildTimeMetricsFields {
+    // The order the enums are defined controls the order the fields are printed in the CSV
+    INITIAL_BUILD_TIME("Initial Build Time", d -> formatDuration(d.getInitialBuildTime())),
+    INSTANT_SAVINGS("Instant Savings", d -> formatDuration(d.getInstantSavings())),
+    INSTANT_SAVINGS_BUILD_TIME("Instant Savings Build Time", d -> formatDuration(d.getInstantSavingsBuildTime())),
+    PENDING_SAVINGS("Pending Savings", d -> formatDuration(d.getPendingSavings())),
+    PENDING_SAVINGS_BUILD_TIME("Pending Savings Build Time", d -> formatDuration(d.getPendingSavingsBuildTime())),
+    ;
+
+    public final String label;
+    public final Function<BuildTimeMetrics, String> value;
+
+    BuildTimeMetricsFields(String label, Function<BuildTimeMetrics, String> value) {
+        this.label = label;
+        this.value = value;
+    }
+
+    public static Stream<BuildTimeMetricsFields> ordered() {
```

Comment:
Returning a Stream is a bit odd, I find. I would return a `List` instead.

Reply:
This is [the same pattern used in the other "Fields" class](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/3870d12078a7410dceb439b53085ebed6cb63c98/components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/BuildValidationDataFields.java#L48-L50). I kept it here for consistency. Would you like me to change it there too?

Though, I'm curious why you find it odd? There are many Java built-in methods that return a `Stream<..>` (e.g. `String#lines`). This simplifies the common use case of callers wanting to stream the enum values. Perhaps, the method could be named differently?

Reply:
Let's keep it here for now.

I find a Stream not great here because it does not communicate anything about the data structure.